server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -738,10 +727,6 @@ void executeBulk(
             this.indexNameExpressionResolver = indexNameExpressionResolver;
         }
 
-        Index getConcreteIndex(String indexOrAlias) {
```

Comment:
After removing the above two usages, this method can be removed as it is no longer used.